### DISCUSSION TITLE
Best way to access user with the new @supabase/ssr package?

### DISCUSSION THREAD

**Discussion by rubenvereecken** [supabase/supabase#26884] [Questions]
I've got @supabase/ssr set up and working – cookies are being set and read nicely, etc. I'm using Next.js.

Now, what's the best way to get the user's session and data in a client component?

Some approaches I see –
1. `useUser` from `@supabase/auth-helpers-react`. This works, but loads with `null` first, making me think it's not accessing the data passed by the server properly (or it's doing so ineffeciently).
2. `await supabase.auth.getUser()` in a server component (because it's async) and somehow pass this along to a child client component

What's the recommended approach?

**Comment by GaryAustin1** [comment]
Good discussion on using getUser and jwt decode approach here:
https://github.com/orgs/supabase/discussions/23224

**Comment by rubenvereecken** [comment]
Thanks for weighing in @GaryAustin1. In summary, is the best way to decode the JWT and cache that?

Would there be no simpler way to access user identity (and have it available without being `null` first) in React?

**Reply by GaryAustin1** [reply]
It would be best to access in that thread.  I don't use SSR.
